package/src/components/MessageList/MessageList.tsx

```diff
@@ -612,14 +612,17 @@ const MessageListWithContext = <
       const isLatestMessageSetShown = !!channel.state.messageSets.find(
         (set) => set.isCurrent && set.isLatest,
       );
-      const msg = processedMessageList?.[messageArrayIndex];
+
       if (!isLatestMessageSetShown) {
+        const msg = processedMessageList?.[messageArrayIndex];
         if (
           channel.state.latestMessages.length !== 0 &&
           unreadCount > channel.state.latestMessages.length
         ) {
           return messageArrayIndex <= unreadCount - channel.state.latestMessages.length - 1;
-        } else if (lastRead && msg.created_at) {
+        }
+        // `msg` can be undefined here, hence we first check if its defined. Eg: `messageArrayIndex` is equal to processedMessageList.length.
```

Comment:
> `messageArrayIndex` is equal to processedMessageList.length.

This can never happen here as its under renderItem

Reply:
I think processedMessageList goes undefined while rendering happens or is changed